File: dev-docs/snippets/gemini/generate.swift
import JsBaoClient

// Send a structured prompt to Gemini and return the generated response.
func generate(client: JsBaoClient) async throws {
  // #region example
  let result = try await client.gemini.generate(options: GeminiGenerateOptions(
    model: "gemini-1.5-pro",
    system: [.text("You are a concise assistant.")],
    messages: [
      GeminiMessage(role: .user, parts: [.text("Summarize photosynthesis.")]),
    ],
    generationConfig: GeminiGenerationConfig(temperature: 0.4, maxOutputTokens: 256)
  ))
  let message = result.message
  // #endregion example
  _ = (result, message)
}

File: dev-docs/snippets/gemini/generate.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Send a structured prompt to Gemini and return the generated response.
export async function generate(client: JsBaoClient) {
  // #region example
  const result = await client.gemini.generate({
    model: "gemini-1.5-pro",
    system: [{ type: "text", text: "You are a concise assistant." }],
    messages: [
      { role: "user", parts: [{ type: "text", text: "Summarize photosynthesis." }] },
    ],
    generationConfig: { temperature: 0.4, maxOutputTokens: 256 },
  });
  // #endregion example
  return result;
}
